### ISSUE TITLE
HttpGetSchemaMiddleware AllowSynchronousIO error

### ISSUE THREAD

**Issue by tunurgitr** [ChilliCream/hotchocolate#1117]
**Describe the bug**
In 10.1, with dotnet core 3, using `HttpGetSchemaMiddleware`, I get the following exeption:
```
System.InvalidOperationException: Synchronous operations are disallowed. Call WriteAsync or set AllowSynchronousIO to true instead.
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http.HttpResponseStream.Flush()
   at System.IO.StreamWriter.Flush(Boolean flushStream, Boolean flushEncoder)
   at System.IO.StreamWriter.Dispose(Boolean disposing)
   at System.IO.TextWriter.Dispose()
   at HotChocolate.AspNetCore.HttpGetSchemaMiddleware.InvokeAsync(HttpContext context)
   at HotChocolate.AspNetCore.QueryMiddlewareBase.InvokeAsync(HttpContext context)
   at HotChocolate.AspNetCore.QueryMiddlewareBase.InvokeAsync(HttpContext context)
   at Microsoft.AspNetCore.Builder.Extensions.MapMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Builder.Extensions.MapMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Builder.Extensions.MapMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Builder.Extensions.MapMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Builder.Extensions.MapMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Builder.Extensions.MapMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Builder.Extensions.MapMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Authentication.AuthenticationMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Builder.Extensions.MapWhenMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Builder.Extensions.MapWhenMiddleware.Invoke(HttpContext context)
```
**Desktop (please complete the following information):**
 - OS: Windows 10, also running on docker `mcr.microsoft.com/dotnet/core/aspnet:3.0-alpine`
 - Version 10.1

**Comment by nigel-sampson** [comment]
We're in the process of moving some of our services to .NET Core 3 and running to this issue as well. 

Updating `SchemaSyntaxSerializer` to be fully asynchronous isn't viable, most of the use cases of `SyntaxVisitor` would be hindered by it being asynchronous.

In the interim we could have `SchemaSyntaxSerializer` write synchronously to an in memory stream and write that asynchronously to the response.

**Comment by michaelstaib** [comment]
@nigel-sampson  Yes, this could be a quick fix. I want to rework the visitors for V11 but in the meantime this could be a good solution. Do you want to implement that? I have another fix already in the version_10_0_0 master so we could create 10.2.1 with these two fixes.

**Comment by nigel-sampson** [comment]
Yup, I'll handle it sometime in the next few days.

**Comment by michaelstaib** [comment]
This issue is fixed with 10.4
